chia/consensus/blockchain.py

```diff
@@ -494,108 +486,109 @@ async def _reconsider_peak(
         if genesis and peak is not None:
             return [], None
 
-        if peak is not None:
```

Comment:
This part is a bit concerning. I'm not sure why, but the original version has two transaction wrappers in this one method, as if this method is non-atomic. My instinct suggests that that whole thing should be an atomic operation. But for now I've kept the split. It doesn't much help that I don't have a good handle on exactly what's going on here.